crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1909,65 +1913,40 @@ impl SymbolicationActor {
 
     /// The actual stackwalking procedure.
     ///
-    /// This is a method, as opposed to a closure in
-    /// [`stackwalk_minidump_with_cfi`](SymbolicationActor::stackwalk_minidump_with_cfi),
-    /// to circumvent a problem with `rustfmt`.
+    /// It can be switched between using the old and new stackwalking methods
+    /// with the `use_new_method` parameter.
     fn procspawn_inner_stackwalk(
         cfi_caches: Json<CfiCacheModules>,
         minidump: Bytes,
-        spawn_time: SystemTime,
-        compare_stackwalking_methods: bool,
+        spawn_time: Option<SystemTime>,
+        use_new_method: bool,
     ) -> Result<Json<StackwalkingOutput>, ProcessMinidumpError> {
         let procspawn::serde::Json(mut cfi_caches) = cfi_caches;
-        if let Ok(duration) = spawn_time.elapsed() {
+        if let Some(duration) = spawn_time.and_then(|st| st.elapsed().ok()) {
```

Comment:
I think this would be nicer if this function did not have to deal with procspawn.  It's a bit unnecessary that this detail ends up in this function.  Practically that means `cfi_caches` and the return types should probably lose their `Json<_>` type wrapper and spawn time can be handled in the closure rather than here as well.

Reply:
Secondly, I think it makes sense to always time the procspawn spawn time.  You could add a tag to say what you're spawning but I don't think there should be a reason to omit the spawn time metric for the new stackwalking code.

Reply:
How should I organize the `minidump.stackwalk.spawn.duration` metrics? If I do `metric!(timer("minidump.stackwalk.spawn.duration"), "method" => "old"/"new")`, which would be my preference, doesn't that mess with the metric's old values?

Reply:
yes, i think what you suggest is good.  i guess you could say you mess with the original metric, but it really is an indication of how heavily the procspawn queue is loaded so that's probably what we want anyway.  also i don't think this is currently part of the main dashboard anyway

Reply:
Now that I think about it, is it even clear that they need to be distinguished? The time measurement happens before the decision on which method to use is even made.

Reply:
i think you might be right, it probably provides very little value and only increases metrics storage cost